src/backend/env/memory.rs

```diff
@@ -179,7 +197,12 @@ impl Default for Allocator {
 }
 
 impl Allocator {
-    fn alloc(&mut self, n: u64) -> Result<u64, String> {
+    fn get_allocation_length(&self, n: u64) -> u64 {
+        (n + self.block_size_bytes - 1) / self.block_size_bytes * self.block_size_bytes
```

Comment:
```suggestion
        (n + self.block_size_bytes - 1) / self.block_size_bytes.pow(2)
```
Where does this formula come from? Also, how do we know that `self.block_size_bytes` is never going to be 0?

Reply:
The suggestion would break the tests because if `/` is an integer division, then `13/4 * 4 = 3 * 4` while `13/16 = 0`.

Reply:
The formula is from me. I just need an upscaling integer division.